### PR TITLE
patron_type: implements checkouts restrictions

### PR DESCRIPTION
**Pull request by zannkukai**
* Adapts the patron_type JSON schema to include a section about borrow
limits.
* Adds some unit tests to validate the schema constraints

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by lauren-d — APPROVED** [review]

**Review by benerken — APPROVED** [review]

File: rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json [hunk 1]
```diff
@@ -81,6 +82,151 @@
...
+          "properties": {
+            "general_limit": {
+              "title": "General checkout limit",
+              "description": "The checkout is blocked once the specified checkout number is reached by the patron. This is a global checkout limit.",
+              "type": "number",
+              "minimum": 1
+            },
+            "library_limit": {
+              "title": "Default checkout limit by library",
+              "description": "The checkout is blocked once the specified checkout number is reached by the patron for items of a same owning library. This is the same limit for all libraries.",
```
Comment (iGormilhit) [inline rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json:115]:
```suggestion
              "description": "The checkout is blocked once the specified checkout number is reached by the patron for items of the same owning library. This is the same limit for all libraries.",
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition (maybe it's wrong and you can do better...):
```
patron_type: implements checkouts restrictions 

* Adapts the patron_type JSON schema to include a section about checkout
  restrictions. There are three type of restrictions: number of checkouts, fee
  amount, and  number of overdue items.
* Adds some unit tests to validate the schema constraints.
```

File: rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json:115]: done

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

File: rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json [hunk 2]
```diff
@@ -81,6 +82,151 @@
...
+      ],
+      "properties": {
+        "checkout_limits": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": [
+            "general_limit"
+          ],
+          "propertiesOrder": [
+            "general_limit",
```
Comment (BadrAly) [inline rero_ils/modules/patron_types/jsonschemas/patron_types/patron_type-v0.0.1.json:102]: `global_limit` or `organisational_limit` I think better than `general_limit` ?

**Review by BadrAly — CHANGES_REQUESTED** [review]

**Review by BadrAly — APPROVED** [review]

**Pull request merged**